compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -485,19 +485,21 @@ OMR::Z::TreeEvaluator::l2aEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    TR_ASSERT(hasCompPtrs, "no compression sequence found under l2a node [%p]\n", node);
    */
 
-   // TODO (GuardedStorage)
-   TR::Register *source;
-   if (comp->useCompressedPointers() && (TR::Compiler->om.compressedReferenceShift() == 0 || firstChild->containsCompressionSequence()) && !node->isl2aForCompressedArrayletLeafLoad())
-      cg->setEvalCompressionSequence(true);
+   TR::Register* source = NULL;
 
-   if ((TR::Compiler->target.cpu.getS390SupportsGuardedStorageFacility()) && comp->useCompressedPointers() && (firstChild->containsCompressionSequence()) && !node->isl2aForCompressedArrayletLeafLoad())
+   // TODO (GuardedStorage): Why is isl2aForCompressedArrayletLeafLoad check necessary?
```

Comment:
> +1 for caution :) . But maybe easier to store the current state before you set it to true, and then restore it to the previous state rather than just set to false? fewer moving parts...

I think I want to keep the depth counter intact to reuse for debugging. I want to place some extra debugging checks to see if the nested scenario happens and if it does in what scenario and how often. We are currently having a lot of trouble debugging issues where we miss to generate a guarded load somewhere in a JIT body and we end up loading a stale pointer during concurrent scavange. These are not easy problems to debug unfortunately as the failure can happen much further down than where the missed guarded load occurred.